medusa/backup.py

```diff
@@ -125,8 +152,16 @@ def _make_manifest_object(self, path_prefix, cached_item):
         return ManifestObject('{}{}'.format(path_prefix, cached_item['path']), cached_item['size'], cached_item['MD5'])
 
     def files_are_different(self, src, cached_item):
+        md5_hash = generate_md5_hash(src)
+        multipart_md5_hash = md5_hash
+        if src.stat().st_size >= int(self._storage_config.multi_part_upload_threshold):
+            multipart_md5_hash = md5_multipart(src, 8)
+
         return (src.stat().st_size != cached_item['size']
-                or (generate_md5_hash(src) != cached_item['MD5'] and self._storage_provider != Provider.LOCAL))
+                or (md5_hash != cached_item['MD5']
```

Comment:
This looks like we could extract all of the hash comparing into a separate function called something like `hashhes_match`.  In that function we could then more comprehensively compare things.

Reply:
I simplified the function, so extracting stuff into a different function would just bring most of it now.